packages/core/src/util/requireHook.js

```diff
@@ -8,25 +8,64 @@
 const Module = require('module');
 const path = require('path');
 
+/**
+ * @typedef {Object} FileNamePatternTransformer
+ * @property {Function} fn
+ * @property {RegExp} pattern
+ */
+
+/**
+ * @typedef {Object} ExecutedHook
+ * @property {*} originalModuleExports
+ * @property {*} moduleExports
+ * @property {Array<string>} appliedByModuleNameTransformers
+ * @property {boolean} byFileNamePatternTransformersApplied
+ */
+
+/**
+ * @typedef {Object.<string, ExecutedHook>} ExecutedHooks
+ */
+
+/**
+ * @type {ExecutedHooks}
+ */
 let executedHooks = {};
+/**
+ * @type {Object.<string, any>}
```

Comment:
We should stay consistent with respect to `any` vs `*` I think. They both have the same semantic, right?

Reply:
I missed this one. Will fix it